### PR TITLE
Removed stuff related to the old bridge

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why is this change needed?

- Provide a description and a link to the underlying ticket

### What changes were made as part of this PR:

- Provide a high level list of the changes made

### Definition of done

- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated 
- [ ] End-to-end tests run if required (see below)

### End-to-end tests

Running the end-to-end tests in the [obscuro-test repo](https://github.com/obscuronet/obscuro-test) is currently at the 
discretion of the developer prior to merging a PR. The intention is not to slow down development by having the longer 
running end-to-end tests as a PR gate run on each branch commit etc. To manually trigger a run pre-merge;

- Go to [run_local_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_local_tests.yml)
- Click the "Run workflow" button
- Use the main branch of obscuro-test 
- Enter the name of the PR branch for go-obscuro
- Run the workflow

The run takes approximately 20 minutes and any failure will be reported in the workflow output. Should the tests fail 
the docker container output for the local testnet and all test artifacts will be added to the run and can be downloaded
for debugging and analysis, with a retention period of 2 days. 

Note that every PR merge to main will also trigger a run of post-merge tests, so even if you do not manually trigger 
pre-merge, tests will be run post-merge. The intention being to catch any issues fast on main to allow for quick 
resolution. The post-merge test output can be seen at 
[run_merge_tests](https://github.com/obscuronet/obscuro-test/actions/workflows/run_merge_tests.yml) and will show both 
the PR number and author in the list of runs.

### CODE REVIEW

File: integration/simulation/validate_chain.go [hunk 1]
```diff
@@ -152,18 +153,14 @@ func checkBlockchainOfEthereumNode(t *testing.T, node ethadapter.EthClient, minH
 		t.Errorf("Node %d: Found Rollup duplicates: %v", nodeIdx, dups)
 	}
 
+	/* TODO: Reenable while old contract deployer is phased out.
```
Comment (tudor-malene) [inline integration/simulation/validate_chain.go:156]: don't follow what the old contract deployer has to do with this

**Review by tudor-malene — CHANGES_REQUESTED** [review]
LGTM overall. 
One change suggestion an a couple of questions

File: go/enclave/bridge/bridge.go [hunk 2]
```diff
@@ -3,12 +3,7 @@ package bridge
 // TODO: once the cross chain messages based bridge is implemented remove this completely.
```
[the remark this answers is not in the document]
Comment (StefanIliev545) [reply go/enclave/bridge/bridge.go]: Changed the package. Some bits are still left like the hardcoded values and the sim test deploying hoc and poc. I'll clean those with https://github.com/obscuronet/obscuro-internal/issues/1158 when everything is migrated to the new bridge.

File: go/enclave/rollupchain/rollup_chain.go [hunk 3]
```diff
@@ -491,30 +490,6 @@ func (rc *RollupChain) processState(batch *core.Batch, txs []*common.L2Tx, state
 		rc.logger.Crit(fmt.Sprintf("Could not retrieve a proof for batch %s", batch.Hash()), log.ErrKey, err)
 	}
 
-	// TODO: Remove this depositing logic once the new bridge is added.
-	depositTxs := rc.bridge.ExtractDeposits(
```
Comment (otherview) [inline go/enclave/rollupchain/rollup_chain.go:495]: The bridge is no longer processing deposits and withdrawals ?

File: integration/simulation/transaction_injector.go [hunk 4]
```diff
@@ -241,101 +239,12 @@ func (ti *TransactionInjector) issueRandomTransfers() {
 
 // issueRandomDeposits creates and issues a number of transactions proportional to the simulation time, such that they can be processed
 func (ti *TransactionInjector) issueRandomDeposits() {
-	var err error
```
Comment (otherview) [inline integration/simulation/transaction_injector.go:244]: If there's no deposits will there be enough obx on the l2 to issue txs ?

**Review by otherview — COMMENTED** [review]
Looks ok. This PR should be split between removing / moving the HOC/POC to a different file and removing withdrawals/deposits from the bridge/sim.

There are a few things here at play and so many changes makes it prone for bugs to slight through

File: integration/simulation/transaction_injector.go (hunk 4 above)
Comment (StefanIliev545) [reply integration/simulation/transaction_injector.go:244]: OBX shouldn't be impacted, but in fairness I should probably check in detail. The bridge is only dealing with HOC and POC, or at least it seems this way.

Comment (otherview) [reply integration/simulation/transaction_injector.go:244]: Yep, these are only for HOC deposits, you're right. The sim injectors are likely using the prefunded keys

File: integration/simulation/network/inmemory.go [hunk 5]
```diff
@@ -37,9 +37,9 @@ func (n *basicNetworkOfInMemoryNodes) Create(params *params.SimParams, stats *st
 
 	// Invent some addresses to assign as the L1 erc20 contracts
 	dummyOBXAddress := datagenerator.RandomAddress()
-	params.Wallets.Tokens[bridge.HOC].L1ContractAddress = &dummyOBXAddress
+	params.Wallets.Tokens[rollupextractor.HOC].L1ContractAddress = &dummyOBXAddress
```
Comment (tudor-malene) [inline integration/simulation/network/inmemory.go:40]: do we still need he hoc and poc hardcoded in the main codebase?

File: go/enclave/rollupchain/rollup_chain.go [hunk 6]
```diff
@@ -46,7 +46,7 @@ type RollupChain struct {
 
 	storage              db.Storage
 	l1Blockchain         *gethcore.BlockChain
-	bridge               *bridge.Bridge
+	bridge               *rollupextractor.RollupExtractor
```
Comment (tudor-malene) [inline go/enclave/rollupchain/rollup_chain.go:49]: these should be renamed as well

File: go/enclave/rollupextractor/rollupextractor.go [hunk 7]
```diff
@@ -68,8 +68,8 @@ type ERC20Mapping struct {
 	L2Address *gethcommon.Address
 }
 
-// Bridge encapsulates all logic around processing the interactions with an L1
-type Bridge struct {
+// RollupExtractor encapsulates all logic around processing the interactions with an L1
+type RollupExtractor struct {
 	// BridgeAddress The address the bridge on the L2
```
Comment (tudor-malene) [inline go/enclave/rollupextractor/rollupextractor.go:73]: comment nnot accurate

File: go/enclave/rollupextractor/rollupextractor.go [hunk 8]
```diff
@@ -68,8 +68,8 @@ type ERC20Mapping struct {
 	L2Address *gethcommon.Address
 }
 
-// Bridge encapsulates all logic around processing the interactions with an L1
-type Bridge struct {
+// RollupExtractor encapsulates all logic around processing the interactions with an L1
```
Comment (tudor-malene) [inline go/enclave/rollupextractor/rollupextractor.go:71]: comment not entirely accurate

**Review by tudor-malene — COMMENTED** [review]
looks good . few qustions

File: integration/simulation/network/inmemory.go (hunk 5 above)
Comment (tudor-malene) [reply integration/simulation/network/inmemory.go:40]: ok, pls add a todo to get rid of them ASAP.
Let's remove all trace of them from the codebase when we can

**Review by tudor-malene — APPROVED** [review]
lgtm

(pls add todo to get rid of hoc/poc from the rollup extractor)

File: integration/simulation/network/inmemory.go (hunk 5 above)
Comment (StefanIliev545) [reply integration/simulation/network/inmemory.go:40]: https://github.com/obscuronet/obscuro-internal/issues/1158 - i've updated this ticket

**Pull request merged**